demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+// Copyright (C) 2018-2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vector>
+#include <queue>
+#include <memory>
+#include <string>
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+#include <cstdio>
+#include <functional>
+#include <atomic>
+
+#include <inference_engine.hpp>
+
+#include <ie_iextension.h>
+
+#include <samples/common.hpp>
+#include <samples/slog.hpp>
+#include <samples/args_helper.hpp>
+#include <samples/ocv_common.hpp>
+
+#include <vpu/vpu_plugin_config.hpp>
+#include <cldnn/cldnn_config.hpp>
+
+#include "imagenet_classification_demo.hpp"
+#include "classification_results.h"
+#include "grid_mat.hpp"
+
+using namespace InferenceEngine;
+
+class IRInfo {
+public:
+    class IRImage {
+    public:
+        cv::Mat mat;
+        unsigned rightClass;
+        long long startTime;
+
+        IRImage(cv::Mat &mat, unsigned rightClass, long long startTime):
+            mat(mat), rightClass(rightClass), startTime(startTime) {}
+    };
+
+    InferRequest &ir;
+    std::vector<IRImage> images;
+
+    IRInfo(InferRequest &ir, std::vector<IRImage> images):
+        ir(ir), images(images) {}
+};
```

Comment:
`IR` is used to refer to `intermediate representation` sometimes. I think the name should be changed.
These classes are simple enough to name them structs.
The constructors aren't needed, you can rely on [aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization).